### PR TITLE
Upgrade base challenge template

### LINKED ISSUE #144: Update speedrun challenge-base branch based on new version of scaffold-stark
## Issue Overview

Scaffold Stark has been updated into version 3 with a lot of new features. We want to make sure that speedrunstark is in sync with the Scaffold Stark version.

## Proposed Solutions or Ideas

- Update the base fork branch
- Resolve any breaking changes

### PR DESCRIPTION
**Pull request by Buuh2511**
# Sync with the Scaffold Stark version

Fixes #144 

## Types of change

- [ ] Feature
- [ ] Bug
- [x] Enhancement

## Comments (optional)
Scaffold Stark has been updated into version 3 with a lot of new features. This PR make sure that speedrunstark is in sync with the Scaffold Stark version

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1,3 @@
+scarb 2.8.2
```
Comment (metalboyrick) [inline .tool-versions:1]: @jrcarlos2000 @Nadai2010 are we committing this file?

**Review by metalboyrick — COMMENTED** [review]
so far looks correct

**Comment by metalboyrick** [comment]
any errors in particular that you have doubts on? @Buuh2511

File: .tool-versions (hunk 1 above)
Comment (Nadai2010) [reply .tool-versions:1]: I think it is useful, at least the version of Scarb and starknet-devnet, can commit IMO

File: packages/nextjs/scaffold.config.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ export type ScaffoldConfig = {
 };
 
 const scaffoldConfig = {
-  targetNetworks: [chains.devnet],
+  targetNetworks: [chains.sepolia],
```
Comment (Nadai2010) [inline packages/nextjs/scaffold.config.ts:12]: Always leave in devnet @Buuh2511

File: packages/snfoundry/contracts/src/test/TestContract.cairo
Comment (Nadai2010) [inline packages/snfoundry/contracts/src/test/TestContract.cairo:1]: The TestContract.cairo must be updated, the tests fail, check new version

https://github.com/Scaffold-Stark/scaffold-stark-2/blob/main/packages/snfoundry/contracts/src/test/TestContract.cairo

**Review by Nadai2010 — COMMENTED** [review]
Revise 2 comments

**Review by gianalarcon — APPROVED** [review]

**Review by metalboyrick — APPROVED** [review]
smoke test looks good

**Pull request merged**